src/interaction/command/circle.ts

```diff
@@ -0,0 +1,260 @@
+      subcommand.setName("repost");
+      subcommand.setDescription("Repost the circle embeds");
+      return subcommand;
+    });
+  }
+
+  public async handleInteraction({ bot, interaction }: SlashCommandContext) {
+    const subComm = interaction.options.getSubcommand();
+    switch (subComm) {
+      case "add":
+        await interaction.deferReply();
+        await addCircle(bot, interaction);
+        break;
+      case "create-channel":
+        await interaction.deferReply();
+        await createChannel(bot, interaction);
+        break;
+      case "repost":
+        await bot.managers.circle.repost();
+        await interaction.reply("Updated circle data");
+    }
+  }
+}
+
+async function createChannel(bot: Bot, interaction: CommandInteraction) {
+  const name = interaction.options.getString("name", true);
+  const circleId = interaction.options.getRole("circle", true).id;
+  const guild = interaction.guild!;
+
+  const circle = bot.managers.database.cache.circles.get(
+    circleId
+  ) as CircleData;
+  if (!circle) {
+    await interaction.editReply({
+      content: "That circle does not exist.",
+    });
+    return;
+  }
+  const channelName = `${circle.emoji} ${name}`;
+  const channel = await guild.channels.create(channelName, {
+    type: "GUILD_TEXT",
+    parent: settings.circles.parentCategory,
+    permissionOverwrites: [
+      {
+        id: guild.roles.everyone,
+        deny: ["VIEW_CHANNEL"],
+        type: "role",
+      },
+      {
+        id: circleId,
+        allow: ["VIEW_CHANNEL"],
+        type: "role",
+      },
+    ],
+  });
+  const newSub = [...circle.subChannels, channel.id];
+  const res = await bot.managers.database.circleUpdate(circleId, {
+    subChannels: newSub,
+  });
+
+  if (!res) {
+    await interaction.editReply({
+      content: "Failed to update database",
+    });
+    return;
+  }
+
+  await interaction.editReply({
+    content: `Created channel ${channel} for ${circle.name}.`,
+  });
+}
+
+async function addCircle(bot: Bot, interaction: CommandInteraction) {
+  const circle: CircleData = {
+    name: interaction.options.getString("name", true),
+    description: interaction.options.getString("description", true),
+    emoji: interaction.options.getString("emoji", true),
+    imageUrl: interaction.options.getString("graphic", true),
+    createdOn: new Date(),
+    owner: interaction.options.getUser("owner", true).id,
+    subChannels: [],
+  };
+  const owner = interaction.options.getMember("owner", true);
+
+  const color = interaction.options.getString("color", true) as ColorResolvable;
+
+  const guild = interaction.guild!;
+  const circleRole = await guild.roles.create({
+    name: `${circle.emoji} ${circle.name}`,
+    mentionable: true,
+    color,
+  });
+  const permissions: OverwriteResolvable[] = [
+    {
+      id: interaction.guild!.id,
+      deny: ["VIEW_CHANNEL"],
+      type: "role",
+    },
+    {
+      id: circleRole,
```

Comment:
`circleRole.id`? based on the current implementation i guess passing the role itself works...

Reply:
not worth fixing, it works :D